### PR TITLE
feat(labels): add deprecated/demo/beta labels to headers + nav items

### PR DESCRIPTION
**Pull request by nicolethoen**
Closes https://github.com/patternfly/patternfly-org/issues/3370
Closes https://github.com/patternfly/patternfly-org/issues/3371
Closes https://github.com/patternfly/patternfly-org/issues/3467
Closes https://github.com/patternfly/patternfly-org/issues/3489

Implements designs from design issue https://github.com/patternfly/patternfly-design/issues/1239

Note: I need to add some frontmatter metadata to a number of core/react to make banners appear at the top of some deprecated tabs. This PR is meant to enable the displaying of labels/banners, and still requires some follow up work in other repos.

### CODE REVIEW

**Review by mcarrano — CHANGES_REQUESTED** [review]
@nicolethoen this looks great! Just two small things to address...

* The design calls for an 8px spacing between the nav names and the labels. Looks like you have more than that.

Screenshot 2023-05-23 at 3 26 56 PM

* I did a search for examples that are currently labeled with the word Legacy. There are two: Navigation - Legacy tertiary (https://www.patternfly.org/v4/components/navigation#legacy-tertiary) and Legacy Spinner (https://www.patternfly.org/v4/components/spinner/html#basic-legacy and https://www.patternfly.org/v4/components/spinner/html#sizes-legacy) Can we mark these examples as deprecated and remove the word "legacy?"

**Review by lboehling — CHANGES_REQUESTED** [review]
thank you @nicolethoen! I also noted the two changes @mcarrano just mentioned above, but other than that this looks great!

**Comment by nicolethoen** [comment]
> * Can we mark these examples as deprecated and remove the word "legacy?"

I've made a note to update these when I make the corresponding changes in core/react as follow up as soon as this merges.

**Review by wise-king-sullyman — APPROVED** [review]

**Review by mcarrano — APPROVED** [review]
Looks great. Thanks @nicolethoen !

**Review by lboehling — APPROVED** [review]
awesome!!

File: packages/documentation-framework/components/example/example.js [hunk 1]
```diff
@@ -173,12 +177,36 @@ export const Example = ({
...
+              )}
+              {isDemo && (
+                <Tooltip content="Demos show how multiple components can be used in a single design.">
+                  <Button variant="plain">
+                    <Label isCompact color="purple">Demo</Label>
+                  </Button>
+                </Tooltip>
+              )}
+              {isDeprecated && (
+                <Tooltip content="Deprecated components are available for use but are no longer being maintained or enhanced.">
```
Comment (evwilkin) [inline packages/documentation-framework/components/example/example.js:202]: Maybe a Surge issue not updating from a previous commit, but I'm seeing slightly different tooltip messages for these three items (Beta, Demo, Deprecated)

Screenshot 2023-05-24 at 4 55 25 PM

Comment (nicolethoen) [reply packages/documentation-framework/components/example/example.js:202]: good catch! i'll fix that

Comment (evwilkin) [reply packages/documentation-framework/components/example/example.js:202]: Sorry - the text in my screenshot above actually maps to the [tooltip in mdx.js](https://github.com/patternfly/patternfly-org/pull/3547/files#diff-da135efaa2b01f501f1ac196874e6c6ae745345196f59a4581445639bf5384c4R253), but curious if this text should match or maybe live in a single variable somewhere that can be referenced?

File: packages/documentation-framework/templates/mdx.js [hunk 2]
```diff
@@ -60,9 +65,19 @@ const MDXChildTemplate = ({
...
       )}
+      {(deprecated || source === 'react-deprecated') && (
+        <InlineAlert title="Deprecated feature" variant="warning">
+          This implementation has been deprecated in favor of a newer implementation, and is no longer getting maintained or enhanced.
+          {newImplementationLink && (
+            <React.Fragment>
+              You can find the <Link to={newImplementationLink}>updated implementation here</Link>.
+            </React.Fragment>
+          )}
+          To learn more about the process, visit our <Link to="/get-started/about#major-release-cadence">about page</Link>.
```
Comment (evwilkin) [inline packages/documentation-framework/templates/mdx.js:76]: FYI there's currently no space between sentences

Screenshot 2023-05-24 at 5 02 37 PM

File: packages/documentation-framework/templates/mdx.js [hunk 3]
```diff
@@ -126,6 +143,9 @@ export const MDXTemplate = ({
   id,
   componentsData
 }) => {
+  const isDeprecated = sources.some(source => source.source === "react-deprecated") && !sources.some(source => source.source === "react");
+  const isBeta = sources.some(source => source.beta)
+  const isDemo = sources.some(source => source.source === "react-demos" || source.source === "html-demos") && !sources.some(source => source.source === "react" || source.source === "html");
```
Comment (evwilkin) [inline packages/documentation-framework/templates/mdx.js:148]: Something we're guilty of throughout docs-framework are these PF-specific hard coded values.  Now that the docs-framework is a package used outside just the PF website build we should try to allow these to be customized by the user whenever possible.

File: packages/documentation-framework/templates/mdx.js [hunk 4]
```diff
@@ -60,9 +65,19 @@ const MDXChildTemplate = ({
           To learn more about the process, visit our <Link to="/get-started/about#beta-components">about page</Link> or our <a href="https://github.com/patternfly/patternfly-org/tree/main/beta-component-promotion">Beta components</a> page on GitHub.
         </InlineAlert>
       )}
+      {(deprecated || source === 'react-deprecated') && (
+        <InlineAlert title="Deprecated feature" variant="warning">
+          This implementation has been deprecated in favor of a newer implementation, and is no longer getting maintained or enhanced.
+          {newImplementationLink && (
```
Comment (evwilkin) [inline packages/documentation-framework/templates/mdx.js:71]: We're not parsing `newImplementationLink` from the markdown anywhere currently so there's no way to trigger this code.  We'd need to add 
```
newImplementationLink: frontmatter.newImplementationLink || false
```
where we're defining `pageData` [in mdx.js (similar to your added code for `deprecated`, `beta`, and `demo`)](https://github.com/patternfly/patternfly-org/pull/3547/files#diff-97bd431987a2e7dda0f66ad7f07ac6cf4eeebf21fbe44a1b45ef905968aa14caR107-R109).

File: packages/documentation-framework/templates/mdx.js (hunk 3 above)
Comment (evwilkin) [reply packages/documentation-framework/templates/mdx.js:148]: FYI this is widespread throughout the package, this works for now but noting for possible future cleanup effort.

**Comment by evwilkin** [comment]
@nicolethoen a few callouts, mostly nits that could be resolved in follow-up issues if needed other than the `newImplementationLink` question

File: packages/documentation-framework/templates/mdx.js (hunk 3 above)
Comment (nicolethoen) [reply packages/documentation-framework/templates/mdx.js:148]: yeah and i'm not sure if I can programmatically determine that a deprecated label can be added to the Example title without some sort of data being manually stored in the component-data. That was going to be my fall back, but this felt like less maintenance for now.

File: packages/documentation-framework/components/example/example.js (hunk 1 above)
Comment (nicolethoen) [reply packages/documentation-framework/components/example/example.js:202]: I made them match. We don't right now have like a constants or strings file. Maybe that's something else we can clean up to add when we clean up the docs-framework in the future.

**Review by wise-king-sullyman — APPROVED** [review]

**Pull request merged**